provider/vsphere/internal/vsphereclient/createvm.go

```diff
@@ -480,6 +486,39 @@ func (c *Client) addNetworkDevice(
 	return &networkDevice, nil
 }
 
+// generateMAC generates a random hardware address that meets VMWare
+// requirements for MAC address: 00:50:56:XX:YY:ZZ where XX is between 00 and 3f.
+func GenerateMAC() (string, error) {
+	c, err := rand.Int(rand.Reader, big.NewInt(4194303))
+	if err != nil {
+		return "", err
+	}
+	r := c.Uint64()
+	return fmt.Sprintf("00:50:56:%.2x:%.2x:%.2x", (r>>16)&0xff, (r>>8)&0xff, r&0xff), nil
+}
+
+// VerifyMAC verifies that the MAC is valid for VMWare.
+func VerifyMAC(mac string) bool {
+	parts := strings.Split(mac, ":")
+	if len(parts) != 6 {
+		return false
+	}
+	if parts[0] != "00" || parts[1] != "50" || parts[2] != "56" {
+		return false
+	}
+	for i, part := range parts[3:] {
+		v, err := strconv.ParseUint(part, 16, 8)
+		if err != nil {
+			return false
+		}
+		if i == 0 && v > 0x3f {
+			// 4th byte must be <= 0x3f
+			return false
+		}
+	}
+	return true
```

Comment:
govmomi seems to say this:
var vmwOUI = net.HardwareAddr([]byte{0x0, 0xc, 0x29})

// From http://pubs.vmware.com/vsphere-60/index.jsp?topic=%2Fcom.vmware.vsphere.networking.doc%2FGUID-DC7478FF-DC44-4625-9AD7-38208C56A552.html
// "The host generates generateMAC addresses that consists of the VMware OUI 00:0C:29 and the last three octets in hexadecimal
//  format of the virtual machine UUID.  The virtual machine UUID is based on a hash calculated by using the UUID of the

at the very least 0xc doesn't seem to match 50.
//  ESXi physical machine and the path to the configuration file (.vmx) of the virtual machine."